### DISCUSSION TITLE
Prism 9: Some questions

### DISCUSSION THREAD

**Discussion by YkTru** [PrismLibrary/Prism#2925] [WPF] [answered]
Just a few questions about Prism 9:

1- Do I have to wait for it or can I use 8 without too many breaking changes when I upgrade?

2a- Will it bring source generator attributes for Observable properties/collection/relaycommand/messenger/viewmodel/validation like with Community toolkit Mvvm or DevExpress.CodeGens?

2b- If not, do you think I can use the Community Toolkit source generator using Prism without conflicts even if it has to inherit from "Observable Object"?

3- Is there an equivalent to "RequestMessage" in the Community Toolkit, where the recipient can request data (and not just wait for notification) indirectly from the sender? If not, is this planned for Prism 9?

4a- I know there was a "logging" extension in previous versions of Prism, do you intend to add such an extension and a "serialization" extension like with Catel?

4b- I find it quite confusing (and so do many others as I can see in various forums) how to implement a serialization service just for layout state; should I create a generic service that all my ViewModels need to implement, then a "LayoutDataModel" record for each control (or one for each VM?) that I instantiate in the VM's ctor to represent the specific layout properties to be saved, etc.?

4c- Are there any "actual best mvvm practices" for serialization ensuring scalability/maintainability (since controls/datatemplate might change often) that you would recommend, or any specific documentation?

Thank you very much!

**Comment by dansiegel** [comment, marked as answer]
1) The API will largely be the same from Prism 8.1 to Prism 9.0. That said there are going to be some binary breaks as we move the interfaces to Prism.Core. For Prism.Forms developers using Regions there may be some small breaks as we will be using an API that is uniform across all platforms. There will also be some breaks with namespaces for Dialogs that will affect all platforms, and likely also for the NavigationParameters as they will be shared between Regions and Prism.Forms Navigation

2) You can use ANY base class you like for your ViewModels whether that's the CommunityToolkit.Mvvm's ObservableObject or ReactiveUI's ReactiveObject or Prism's BindableBase. That said if you are looking for source generators those will ONLY be available to Sponsors and those who support both themselves and Prism with an Enterprise Support Contract or potential future paid Prism license (TBD).

3) If there is a feature which Prism does not currently have we always encourage you to submit a draft spec with information as to what the problem is and why you feel it would benefit the community. We may or may not decide to add it to a future release of Prism, but @brianlagunas and I always encourage the community to actively make suggestions, submit PRs and help drive Prism forward.

4a) I'll most likely continue to ship the Prism.Plugin.Logging group of loggers extensions though they really don't have much to do with Prism at this point beyond having some registration helpers that use Prism's DI abstractions.

**Comment by PabloRuiz-TT** [comment]
What is the data type to be passed to the ObservableObject<t> of prism maui ? thanks
